### PR TITLE
Refactor cancel slurm job

### PR DESCRIPTION
**Pull request by henrikstranneheim**
## Description

Refactor cancel slurm job

### Changed

- Refactor and migrate`cancel_slurm_job` to SLURM API
- Removed `cancel_slurm_job` from Mockstore

### How to prepare for test
- [ ] ssh to hasta (depending on type of change)
- [ ] activate stage: `us`
- [ ] request trailblazer-stage on hasta: `paxa`
- [ ] install on stage:
    ```Shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_trailblazer -t trailblazer -b refactor_cancel_slurm_job -a
    ```
- [ ] ssh to clinical-db (depending on type of change)
- [ ] install on stage:
`bash /home/proj/production/servers/resources/clinical-db.scilifelab.se/update-trailblazer-ui-stage.sh refactor_cancel_slurm_job`

### How to test
- [ ] login to ...
- [ ] do ...

### Expected test outcome
- [ ] check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review
- [ ] tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a
- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: tests/cli/test_cli_core.py [hunk 1]
```diff
@@ -199,6 +199,9 @@ def test_cancel_ongoing_analysis(
         ).decode(CharacterFormat.UNICODE_TRANSFORMATION_FORMAT_8),
     )
 
+    # GIVEN SLURM squeue output for an analysis
```
Comment (henrikstranneheim) [inline tests/cli/test_cli_core.py:202]:
```suggestion
    # GIVEN SLURM scancel output for an analysis
```

File: trailblazer/apps/slurm/api.py [hunk 2]
```diff
@@ -22,6 +22,16 @@ def _get_squeue_jobs_flag_input(slurm_job_id_file_content: Dict[str, List[str]])
     return ",".join(job_ids)
 
 
+def cancel_slurm_job(slurm_id: int, use_ssh: bool = False) -> None:
+    """Cancel SLURM job by SLURM job id."""
+    scancel_commands: List[str] = ["scancel", str(slurm_id)]
+    if use_ssh:
+        scancel_commands = ["ssh", "hiseq.clinical@hasta.scilifelab.se"] + scancel_commands
+        subprocess.Popen(scancel_commands)
+        return
```
Comment (diitaz93) [inline trailblazer/apps/slurm/api.py:31]: I think these lines are not necessary; is there any specific reason to leave them? If there is, feel free to disregard
```suggestion
```

**Review by diitaz93 — APPROVED** [review]
Great👍

File: trailblazer/apps/slurm/api.py (hunk 2 above)
Comment (henrikstranneheim) [reply trailblazer/apps/slurm/api.py:31]: It is a bit tricky to see, but both are used.

**Comment by henrikstranneheim** [comment]
### CLI and API ✅ 

Screenshot 2023-09-05 at 10 35 28

```
68° [henrik.stranneheim@hasta:~] [S_base] 15s $ trailblazer delete  --cancel-jobs --force 278298
Cancelling job 5092208 - bwa_mem2
Cancelling job 5092207 - bwa_mem2
Cancelling job 5092209 - samtools_merge
Cancelling job 5092210 - samtools_merge
Cancelling job 5092211 - samtools_merge
Cancelling job 5092212 - markduplicates
Cancelling job 5092213 - markduplicates
Cancelling job 5092214 - markduplicates
Cancelling job 5092240 - cnvnator_ar
Cancelling job 5092241 - cnvnator_ar
Cancelling job 5092242 - cnvnator_ar
Cancelling job 5092244 - manta
Cancelling job 5092248 - sv_varianteffectpredictor
Cancelling job 5092249 - sv_vcfparser
Cancelling job 5092250 - sv_rankvariant
Cancelling job 5092263 - deepvariant
Cancelling job 5092264 - deepvariant
Cancelling job 5092265 - deepvariant
Cancelling job 5092266 - glnexus_merge
Cancelling job 5092282 - prepareforvariantannotationblock
Cancelling job 5092283 - rhocall_ar
Cancelling job 5092284 - bcftools_norm
Cancelling job 5092285 - variant_annotation
Cancelling job 5092295 - frequency_filter
Cancelling job 5092296 - cadd_ar
Cancelling job 5092297 - varianteffectpredictor
Cancelling job 5092298 - vcfparser_ar
Cancelling job 5092299 - rankvariant
Cancelling job 5092215 - chanjo_sexcheck
Cancelling job 5092216 - chanjo_sexcheck
Cancelling job 5092217 - chanjo_sexcheck
Cancelling job 5092218 - samtools_subsample_mt
Cancelling job 5092219 - samtools_subsample_mt
Cancelling job 5092220 - samtools_subsample_mt
Cancelling job 5092221 - tiddit_coverage
Cancelling job 5092222 - tiddit_coverage
Cancelling job 5092223 - tiddit_coverage
Cancelling job 5092224 - chromograph_cov
Cancelling job 5092225 - chromograph_cov
Cancelling job 5092226 - chromograph_cov
Cancelling job 5092227 - mitodel
Cancelling job 5092228 - mitodel
Cancelling job 5092229 - mitodel
Cancelling job 5092230 - sambamba_depth
Cancelling job 5092231 - sambamba_depth
Cancelling job 5092232 - sambamba_depth
Cancelling job 5092233 - smncopynumbercaller
Cancelling job 5092234 - picardtools_collectmultiplemetrics
Cancelling job 5092235 - picardtools_collectmultiplemetrics
Cancelling job 5092236 - picardtools_collectmultiplemetrics
Cancelling job 5092237 - picardtools_collecthsmetrics
Cancelling job 5092238 - picardtools_collecthsmetrics
Cancelling job 5092239 - picardtools_collecthsmetrics
Cancelling job 5092243 - expansionhunter
Cancelling job 5092245 - tiddit
Cancelling job 5092246 - sv_combinevariantcallsets
Cancelling job 5092247 - sv_annotate
Cancelling job 5092251 - sv_reformat
Cancelling job 5092252 - vcf2cytosure_ar
Cancelling job 5092253 - me_merge_bam
Cancelling job 5092254 - me_merge_bam
Cancelling job 5092255 - me_merge_bam
Cancelling job 5092256 - retroseq
Cancelling job 5092257 - retroseq
Cancelling job 5092258 - retroseq
Cancelling job 5092259 - me_merge_vcfs
Cancelling job 5092260 - me_annotate
Cancelling job 5092261 - me_varianteffectpredictor
Cancelling job 5092262 - me_filter
Cancelling job 5092267 - gatk_combinevariantcallsets
Cancelling job 5092268 - peddy_ar
Cancelling job 5092269 - plink
Cancelling job 5092270 - gatk_variantevalall
Cancelling job 5092271 - gatk_variantevalall
Cancelling job 5092272 - gatk_variantevalall
Cancelling job 5092273 - gatk_collectreadcounts
Cancelling job 5092274 - gatk_collectreadcounts
Cancelling job 5092275 - gatk_collectreadcounts
Cancelling job 5092276 - gatk_denoisereadcounts
Cancelling job 5092277 - gatk_denoisereadcounts
Cancelling job 5092278 - gatk_denoisereadcounts
Cancelling job 5092279 - gens_generatedata
Cancelling job 5092280 - gens_generatedata
Cancelling job 5092281 - gens_generatedata
Cancelling job 5092286 - rhocall_viz
Cancelling job 5092287 - rhocall_viz
Cancelling job 5092288 - rhocall_viz
Cancelling job 5092289 - chromograph_rhoviz
Cancelling job 5092290 - chromograph_rhoviz
Cancelling job 5092291 - chromograph_rhoviz
Cancelling job 5092292 - upd_ar
Cancelling job 5092293 - chromograph_upd
Cancelling job 5092294 - mt_annotation
Cancelling job 5092300 - endvariantannotationblock
Cancelling job 5092301 - gatk_variantevalexome
Cancelling job 5092302 - gatk_variantevalexome
Cancelling job 5092303 - gatk_variantevalexome
Cancelling job 5092304 - qccollect_ar
Cancelling job 5092305 - version_collect_ar
Cancelling job 5092306 - multiqc_ar
Cancelling job 5092307 - analysisrunstatus
Cancelling job 5092308 - sacct
Cancelling job 5092206 - bwa_mem2
Cancelling job 5092205 - fastqc_ar
Cancelling job 5092203 - fastqc_ar
Cancelling job 5092204 - fastqc_ar
Case justhusky - Analysis 278298: all ongoing jobs cancelled successfully!
Updated status justhusky - 278298: running
Deleting analysis 278298 for case justhusky
```

Screenshot 2023-09-05 at 10 47 47

**Comment by henrikstranneheim** [comment]
Deployed to production:
```
Log deploy... done.
trailblazer, version 17.0.9
θ68° [henrik.stranneheim@hasta:~] [P_base] 1m47s $
```

**Pull request merged**